src/ShopBundle/objects/db/TShopCategory.class.php

```diff
@@ -486,29 +486,37 @@ protected function NumberOfArticlesInCategory($bRecursive = false)
      */
     public function AllowDisplayInShop()
     {
-        $bShowCat = false;
-
-        $oShop = TdbShop::GetInstance();
-        if (!is_null($oShop) && !$oShop->fieldShowEmptyCategories) {
-            $bShowCat = ($this->GetNumberOfArticlesInCategory(true) > 0);
-        } else {
-            $bShowCat = true;
+        if (true === $this->isEmptyAndShouldNotBeShown()) {
+            return false;
         }
 
         if (false == $this->fieldActive || false == $this->fieldTreeActive) {
-            $bShowCat = false;
+            return false;
```

Comment:
since this check is much faster than the "is empty" check, I would suggest moving it up.

Reply:
ok